### ISSUE TITLE
Files and images not syncing

### ISSUE THREAD

**Issue by theced7** [turtl/api#16]
I'm running Turtl in [this](https://github.com/ArthurGarnier/turtl-docker) docker. Text sync completely fine, however, images and files don't. Whenever I'm trying to open a sync'd file I get this error: "There was a problem opening that file: undefined". 

Wookie seems to be running without an error at first:
```
<NOTICE> [1468784902] turtl - *a :: {"event":"file-upload","data":{"shared":null,"uid":"578bd6e379485a0075000001","client":"desktop-0.6.4","size":"100kb - 1mb"}}
  <INFO> [1468784902] wookie - "PUT /notes/0155fa669b0843eb8d37c4680d8ee2e796c05d89edfe4aebbfec6728e9e35d51d5a6db249fd50d19/file" 200 143
  <INFO> [1468784903] wookie - "GET /sync" 200 4
  <INFO> [1468784903] wookie - "GET /sync" 200 766
  <INFO> [1468784903] wookie - "GET /sync" 200 4
  <INFO> [1468784903] wookie - "GET /sync" 200 766
  <INFO> [1468784903] wookie - "GET /notes/0155fa669b0843eb8d37c4680d8ee2e796c05d89edfe4aebbfec6728e9e35d51d5a6db249fd50d19/file" 200 117
  <INFO> [1468784921] wookie - "GET /notes/0155fa649ff5a861c0ac40b00882a3463cd573acf0e34ed1122e080b4e55f3e3321285dcb6ce071b/file" 200 117
  <INFO> [1468784934] wookie - "GET /sync" 200 4
  <INFO> [1468784934] wookie - "GET /sync" 200 4
  <INFO> [1468784936] wookie - "GET /notes/0155fa669b0843eb8d37c4680d8ee2e796c05d89edfe4aebbfec6728e9e35d51d5a6db249fd50d19/file" 200 117
```

However, I get some of these later on:
``` 
<INFO> [1468784996] wookie - "GET /sync" 200 4
 <DEBUG> [1468784996] turtl - turtl error: #<SOCKET-EOF #<TCP-SOCKET #x3020028DBADD>: 0: NIL #x302002B9EF8D>
 <DEBUG> [1468784996] turtl - turtl error: #<SOCKET-EOF #<TCP-SOCKET #x3020028DC39D>: 0: NIL #x302002B8AFCD>
``` 

config.lisp:
```
root@3536a82770d7:/opt/api/config# cat config.lisp
(in-package :turtl)

(defparameter *root* (asdf:system-relative-pathname :turtl #P"")
  "Defines the root directory turtl is loading from (basically the ASDF path).")

(defparameter *pid-file* nil
  "File to write the current proc's PID to.")

(defvar *server-bind* nil
  "The address to bind Turtl to (nil is the same as 0.0.0.0).")

(defvar *server-port* 8181
  "The port we want to start the Turtl service on.")

(defvar *db-name* "turtl"
  "The name of the database we'll be using LOL")

(defvar *db-host* "127.0.0.1"
  "The database hostname/ip.")

(defvar *db-port* 28015
  "The database port number.")

(defvar *production-error-handling* nil
  "If t, will attempt to catch all errors that make it to top-level and not let
   the parent process enter the debugger.")

(defvar *enable-hsts-header* nil
  "If NIL, Turtl won't pass back an HSTS security header. If this is set, it
   should be set to a integer value, which will be passed to the max-age value
   of the header.")

(defvar *enabled-cors-resources* "resource://turtl-at-lyonbros-dot-com"
  "When set, will enable CORS for resource:// origins if they match the given
   string. Entries should be comma separated (this string is passed verbatim in
   the Access-Control-Allow-Origin header).")

(defvar *site-url* "MYDOMAIN"
  "The main URL the site will load from.")

(defvar *api-path* ""
  "The path the API lives under. Can be blank.")

(defvar *admin-email* "your@mom.com"
  "The email used for admin communications. This is reported to users on a
   server error, and possibly other instances.")

(defvar *email-from* "noreply@yourdomain.com"
  "The email address all turtl emails come from.")

(defvar *email-user* ""
  "The username used for sending email. Needs to be set on load.")
  (defvar *email-pass* ""
    "The password used for sending email. Needs to be set on load.")

(defvar *display-errors* nil
  "Whether or not to show errors in HTTP responses. Useful for debugging, bad
   for production.")

(defparameter *public-actions*
  `((:post . ,(concatenate 'string *api-path* "/users"))
    (:post . ,(concatenate 'string *api-path* "/log/error"))
        (:post . "/cla/sign")
            (:get . ,(cl-ppcre:create-scanner (concatenate 'string *api-path* "/invites/codes/([0-9a-f-]+)"))))
  "A list of public resources/actions that do not require authentication.")

(defparameter *default-storage-limit* nil
  "The max amount of data a profile can hold (in megabytes). Set to nil to allow
   infinite size profiles.")

(defparameter *storage-invite-credit* 25
  "The amount of storage (in mb) to credit a user when they refer someone.")

;; setup the logger
(vom:config :turtl :info)

(defvar *analytics* '(:enabled t
                      :db "analytics")
                        "Holds analytics config")

;; -----------------------------------------------------------------------------
;; File storage section.
;; -----------------------------------------------------------------------------

;; Choose either local uploads or S3.

(defvar *local-upload* "/opt/files"
  "NIL disables local files storage (files are uploaded to S3). Set to a local
   path to save files locally instead of remotely. No trailing slash!")
   (defvar *local-upload-url* "MYDOMAIN"
     "Define the URL that local files will be loaded from. Generally, this will be
   the same URL the API is accessed from and only needs to be defined if the
   *local-upload* variable has a value. This should *not* include the /files
   path, and there should be no trailing slash. Example:

     http://turtl.dev:8181")

(defvar *amazon-s3* '(:token ""
                      :secret ""
                      :bucket ""
                      :endpoint "https://s3.amazonaws.com")
  "Holds Amazon S3 config.")
```

Any idea what the problem might be?

**Comment by orthecreedence** [comment]
A few things:

- make sure the app has permissions to write to `/opt/files` (set in `*local-upload*`)
- make sure you specify a value for `*local-upload-url*`. the value is currently `MYDOMAIN`. this should be set to the URL you access the API from.

**Comment by szimszon** [comment]
Hi, I had similar issue I had to define *local-upload-url* and everything is fine now...
Thanks.

**Comment by nukedone** [comment]
I'm sorry for addressing this issue again. I have installed the api from the same docker image. Defined the local-upload-url as "http://192.168.1.100:8181", local-upload to "/opt/files", but still the files added by the desktop client are not visible on the Android device: error openning that file: undefined.

I ran it without the Reverse proxy.
Tried to edit the container with
sudo docker exec -it turtl bash
and then create path /opt/files as well as change the config in config/config.lisp
when I restart the container and edit it again the changes are not preserved.
Don't know if the path in local-upload needs to be on the host or in the container.
 If anyone can help to start using this great app to its full.
